Firestore/core/src/firebase/firestore/core/filter.h

```diff
@@ -67,8 +77,30 @@ class Filter {
 
   /** A unique ID identifying the filter; used when serializing queries. */
   virtual std::string CanonicalId() const = 0;
+
+  /** A debug description of the Filter. */
+  virtual std::string ToString() const = 0;
+
+  virtual size_t Hash() const = 0;
+
+  virtual bool IsInequality() const {
```

Comment:
Optional nit: all other virtual functions in this interface are pure virtuals, perhaps make this one as well, for consistency?